src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -87,30 +75,25 @@ private ITaskItem GetRunInformation()
 
             ProjectInstance projectInstance = new ProjectInstance(Project, globalProperties, null);
 
-            BuildRequestData buildRequestData = new BuildRequestData(projectInstance, new string[] { GetRunInformationTaskName });
-            BuildParameters buildParameters = new BuildParameters();
-
-            BuildResult result = BuildManager.DefaultBuildManager.Build(buildParameters, buildRequestData);
-
-            TargetResult runInfoResult;
-            if (!result.ResultsByTarget.TryGetValue(GetRunInformationTaskName, out runInfoResult))
+            string runProgram = projectInstance.GetPropertyValue(RunCommandPropName);
+            if (string.IsNullOrEmpty(runProgram))
             {
-                throw new InvalidOperationException($"Could not find a target named '{GetRunInformationTaskName}' in your project. Please ensure 'dotnet run' supports this project.");
+                throw new InvalidOperationException($"The property named '{RunCommandPropName}' does not have a value in your project. Please ensure 'dotnet run' supports this project.");
             }
 
-            if (runInfoResult.ResultCode != TargetResultCode.Success)
-            {
-                throw new InvalidOperationException($"Could not get the run information for project {Project}. An internal MSBuild error has occured" + Environment.NewLine +
-                    runInfoResult.Exception?.ToString());
-            }
+            string runArguments = projectInstance.GetPropertyValue(RunArgumentsPropName);
+            string runWorkingDirectory = projectInstance.GetPropertyValue(RunWorkingDirectoryPropName);
 
-            ITaskItem runInfoItem = runInfoResult.Items.FirstOrDefault(i => i.ItemSpec == projectInstance.FullPath);
-            if (runInfoItem == null)
+            string fullArguments = runArguments;
```

Comment:
We discussed this offline, and since the existing `$(StartArguments)` property doesn't have a separator between arguments other than a space.  Trying to parse a command-line string ourselves into separate args, and then put them back into a command-line string doesn't seem like a good approach.

So we will just take the string as it is given to us by MSBuild.  This behavior appears to be consistent with VS F5 - if you have a trailing backslash followed by a double quote, you get a double quote in your args.